src/Data/Digit/Properties.agda

```diff
@@ -0,0 +1,36 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of digits and digit expansions
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Data.Digit
+import Data.Char.Properties as Charₚ
+open import Data.Nat.Base using (ℕ)
+open import Data.Nat.Properties using (_≤?_)
+open import Data.Fin.Properties using (inject≤-injective)
+open import Data.Product using (_,_; proj₁)
+open import Data.Vec.Relation.Unary.Unique.Propositional using (Unique)
+import Data.Vec.Relation.Unary.Unique.Propositional.Properties as Uniqueₚ
+open import Data.Vec.Relation.Unary.AllPairs using (allPairs?)
+open import Relation.Nullary.Decidable using (True; from-yes)
+open import Relation.Nullary.Negation using (¬?)
+open import Relation.Binary.PropositionalEquality as P using (_≡_; refl)
+open import Function using (_∘_)
+
+module Data.Digit.Properties where
+
+digitCharsUnique : Unique digitChars
+digitCharsUnique = from-yes (allPairs? (λ x y → ¬? (x Charₚ.≟ y)) digitChars)
+
+module _ (base : ℕ) where
+  module _ {base≥2 : True (2 ≤? base)} where
```

Comment:
Technically it is not necessary for the `base≥2` and `base≤16` proofs to
be equal on both sides for this to be true. I wonder whether it's worth
generalising it.

Or to prove that for `toDigits` and `showDigit` these arguments are irrelevant.

Reply:
Good catch! I prefer the former, I find it more ergonomic to use. (Also, for the latter, could we not just use the fact that all lte proofs are equal?) I've updated the PR accordingly, making `inject≤-injective` accept two distinct lte proofs as well.